docs/proposals/ha_compact.md

```diff
@@ -0,0 +1,88 @@
+# Thanos Compact - High Availability
+
+Status: **draft** | in-review | rejected | accepted | complete
+
+Implementation Owner: [@domgreen](https://github.com/domgreen)
+
+## Summary
+
+Thanos Compact currently runs as a singleton against a given bucket, this proposal outlines how we could create a highly avaliable
+setup so that given one compactor dies or is unable to reach a bucket another instance of `thanos compact` can take over the data compaction.
+
+## Motivation
+
+If a `thanos compact` instance dies we would like another to take over compaction of the data in a bucket.
+
+## Goals
+
+- ensure `thanos compact` can be run in a HA manner
+
+## Proposal
+
+An initial proposal of adding ETCD into `thanos compact` so that we can run master election against multiple components.
```

Comment:
worth to mention exact client code for this? https://github.com/etcd-io/etcd/blob/master/clientv3/concurrency/election.go#L44